test/unit/services/test_catalogue_category.py

```diff
@@ -217,6 +219,62 @@ def test_create_with_leaf_parent_catalogue_category(
     assert str(exc.value) == "Cannot add catalogue category to a leaf parent catalogue category"
 
 
+def test_create_with_duplicate_property_names(
+    test_helpers, catalogue_category_repository_mock, catalogue_category_service
+):
+    """
+    Test trying to create a catalogue category with duplicate catalogue item property names
+    """
+    # pylint: disable=duplicate-code
+    catalogue_category = CatalogueCategoryOut(
+        id=str(ObjectId()),
+        name="Category B",
+        code="category-b",
+        is_leaf=True,
+        parent_id=str(ObjectId()),
+        catalogue_item_properties=[
+            CatalogueItemProperty(name="Property A", type="number", unit="mm", mandatory=False),
+            CatalogueItemProperty(name="Property A", type="boolean", mandatory=True),
+        ],
+    )
+    # pylint: enable=duplicate-code
+
+    # Mock `get` to return the parent catalogue category
+    # pylint: disable=duplicate-code
+    test_helpers.mock_get(
+        catalogue_category_repository_mock,
+        CatalogueCategoryOut(
+            id=catalogue_category.parent_id,
+            name="Category A",
+            code="category-a",
+            is_leaf=False,
+            parent_id=None,
+            catalogue_item_properties=[],
+        ),
+    )
+    # pylint: enable=duplicate-code
+
+    # the same catalogue item properties but mapped to the schema so that they are accepted
+    catalogue_category_catalogue_item_properties = [
+        CatalogueItemPropertySchema(name="Property A", type="number", unit="mm", mandatory=False),
+        CatalogueItemPropertySchema(name="Property A", type="boolean", mandatory=True),
+    ]
+
+    with pytest.raises(DuplicatePropertyName) as exc:
+        # pylint: disable=duplicate-code
+        catalogue_category_service.create(
+            CatalogueCategoryPostRequestSchema(
+                name=catalogue_category.name,
+                is_leaf=catalogue_category.is_leaf,
+                catalogue_item_properties=catalogue_category_catalogue_item_properties,
```

Comment:
You can do what you did in `test_update_properties_to_have_duplicate_names` and get rid of the `catalogue_category_catalogue_item_properties` var that way
```suggestion
                catalogue_item_properties=[prop.model_dump() for prop in catalogue_category.catalogue_item_properties],
```